apps/nova/src/components/dashboard/PastLessons.tsx

```diff
@@ -102,22 +112,46 @@ export const PastLessons: React.FC = () => {
           : intl("tutor-dashboard.past-lessons.title")}
       </Typography>
 
-      <PastLessonsTable
-        tutorsRoute={Route.Tutors}
-        onRebook={openRebookingDialog}
-        lessons={lessons}
-        loading={lessonsQuery.query.isPending}
-        error={lessonsQuery.query.isError}
-        retry={lessonsQuery.query.refetch}
-        isTutor={
-          user?.role === IUser.Role.Tutor ||
-          user?.role === IUser.Role.TutorManager
-        }
-        onSendMessage={onSendMessage}
-        sendingMessage={sendingMessage}
-      />
-
-      {!lessonsQuery.query.isFetching && lessonsQuery.query.hasNextPage ? (
+      {mq.lg ? (
+        <PastLessonsTable
+          tutorsRoute={Route.Tutors}
+          onRebook={openRebookingDialog}
+          lessons={lessons}
+          loading={lessonsQuery.query.isPending}
+          error={lessonsQuery.query.isError}
+          retry={lessonsQuery.query.refetch}
+          isTutor={
+            user?.role === IUser.Role.Tutor ||
+            user?.role === IUser.Role.TutorManager
+          }
+          onSendMessage={onSendMessage}
+          sendingMessage={sendingMessage}
+        />
+      ) : (
+        <PastLessonsSummary
+          tutorsRoute={Route.Tutors}
+          onRebook={openRebookingDialog}
+          lessons={lessons}
+          loading={lessonsQuery.query.isPending}
+          error={lessonsQuery.query.isError}
+          retry={lessonsQuery.query.refetch}
+          isTutor={
+            user?.role === IUser.Role.Tutor ||
+            user?.role === IUser.Role.TutorManager
+          }
+          onSendMessage={onSendMessage}
+          sendingMessage={sendingMessage}
+          more={() => {
+            if (mq.lg || !lessonsQuery.query.hasNextPage) return;
+            if (lessonsQuery.query.hasNextPage) lessonsQuery.more();
+          }}
+          loadingMore={lessonsQuery.query.isFetching}
```

Comment:
You need a prop to disable the more button incase of no more pages. 

```suggestion
          hasMore={lessonsQuery.query.hasNextPage}
          more={() => {
            if (lessonsQuery.query.hasNextPage) lessonsQuery.more();
          }}
          loadingMore={lessonsQuery.query.isFetching}
```